packages/layer-composer/src/layer-composer.ts

```diff
@@ -201,6 +201,7 @@ class LayerComposer {
       return promise
         .then(({ config, style }) => {
           const layer = this._applyGenericStyle(config, style)
+          const metadataStyle = this._getGeneratorMetadata([...layersGenerated, layer])
```

Comment:
Fixes a bug when using async generators but the metadata wasn't properly updated